Sources/NIODependencies/NIO.swift

```diff
@@ -0,0 +1,16 @@
+import Dependencies
+import NIOCore
+import NIOEmbedded
+
+extension DependencyValues {
+  public var mainEventLoopGroup: any EventLoopGroup {
+    get { self[MainEventLoopGroupKey.self] }
+    set { self[MainEventLoopGroupKey.self] = newValue }
+  }
+}
+
+public enum MainEventLoopGroupKey: TestDependencyKey {
+  public static var testValue: any EventLoopGroup {
+    EmbeddedEventLoop()
```

Comment:
NIOCore does not contain the live event loop implementations. To get those we need to compile all of NIO. There's a lightweight `NIOEmbedded` module, though, that we can use in tests, instead.

If we're worried about losing the "integration" of using a live loop, we either need to depend on all of NIO here, or we need to make sure all our tests `withDependencies` a live loop in...